src/nanoemoji/glue_together.py

```diff
@@ -1,37 +1,82 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Copies color assets from one font to another.
+
+Both must use the same glyph names."""
+from absl import app
+from absl import flags
+from absl import logging
 from fontTools.ttLib.tables import otTables as ot
 from fontTools import ttLib
 import os
 
-_ORIGINAL="/tmp/Bungee7.ttf"
-_COLR="/tmp/bungee7/AnEmojiFamily.ttf"
 
-font = ttLib.TTFont(_ORIGINAL)
-colr = ttLib.TTFont(_COLR)
-_OUTPUT = '/tmp/bungee7/Bungee_SVG_COLR.ttf'
+FLAGS = flags.FLAGS
+
+
+flags.DEFINE_string("target_font", None, "Font assets are copied into.")
+flags.DEFINE_string("donor_font", None, "Font from which assets are copied.")
+flags.DEFINE_string("color_table", None, "The color table to copy.")
+flags.DEFINE_string("output_file", None, "Font assets are copied into.")
+
+
+def _copy_colr(target: ttLib.TTFont, donor: ttLib.TTFont):
+    # Copy all glyphs used by COLR over
+    _glyphs_to_copy = set()
+    new_glyphorder = list(target.getGlyphOrder())
+
+    def _collect_glyphs(paint):
+        if paint.Format == ot.PaintFormat.PaintGlyph:
+            _glyphs_to_copy.add(paint.Glyph)
+
+    for record in donor["COLR"].table.BaseGlyphList.BaseGlyphPaintRecord:
+        record.Paint.traverse(donor["COLR"].table, _collect_glyphs)
+
+    for glyph_name in _glyphs_to_copy:
+        print("Copy glyph", glyph_name)
+        target["glyf"].glyphs[glyph_name] = donor["glyf"].glyphs[glyph_name]
+        target["hmtx"].metrics[glyph_name] = donor["hmtx"].metrics[glyph_name]
 
-# Copy all glyphs used by COLR over
-_glyphs_to_copy = set()
+        # .notdef in particular likely to be in both
+        if glyph_name not in new_glyphorder:
+            new_glyphorder.append(glyph_name)
 
-def _collect_glyphs(paint):
-  if paint.Format == ot.PaintFormat.PaintGlyph:
-    _glyphs_to_copy.add(paint.Glyph)
+    target.setGlyphOrder(new_glyphorder)
+    # glyf fails internal checks if not advised of new glyph orderings
```

Comment:
the reason is glyf table borrows a mutable reference of the TTFont.glyphOrder (because TTFont.getGlyphOrder() returns a list), but calling `TTFont.setGlyphOrder()` only replaces the `TTFont.glyphOrder` internal list, leaving the glyf.glyphOrder unchanged... Not ideal, I know.

But look, glyf table has `__setitem__` which calls `self.glyphOrder.append(...)`, which modifies both it's own and the global glyphOrder (they are the same list), maybe you can use that instead.
`target["glyf"][glyph_name] = donor["glyf"][glyph_name]` -- without accessing the interal `glyf.glyphs` attribute.
Technically, it would be preferable to use the glyf table `__getitem__` instead of its internal `glyphs` dict, because the former [expands lazy-loaded glyphs](https://github.com/fonttools/fonttools/blob/ac4d5611ac793e4815f64eb05d46e0cb46ec4612/Lib/fontTools/ttLib/tables/_g_l_y_f.py#L293).